src/Core/Domain/Product/QueryResult/ProductStock.php

```diff
@@ -0,0 +1,218 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\QueryResult;
+
+use DateTime;
+
+class ProductStock
```

Comment:
ProductStockSettings ? ProductStockInformations ?

Reply:
Ok for this one I admit I'm a bit more lost Because the other query results don't have a clear convention:
- `ProductSeoOptions`
- `ProductPricesInformation`
- `ProductSupplierOptions` / `ProductSupplierOption`
- `ProductCategoriesInformation`

It's kind of a mess and would deserve to clarify the naming, either use Options or Informations or nothing But we should do the same everywhere So I suggest we clarify and modify this in a later PR 😅

Reply:
I have already issued this some time ago, we decided just to call as it fits. 
as (if Im not wrong) @matks argumented - Options fits more to something we can adjust. Information is more suitable for (well I don't remember :trollface: ) anyway I agree it is still too much space left for interpretation. 
But it doesnt sound right if e.g. you call ProductCategoriesOptions does it? Information sounds better this way.
Although I would vote that we find one suffix +-fitting all cases

Reply:
Ok you convinced me, I went with `ProductStockInformation` I'm not sure the information suffix is the best choice for all other use cases but in this case it felt better as it doesn't contain only configurations or options but actual dynamic values (like quantity)